optimum/exporters/onnx/convert.py

```diff
@@ -540,6 +540,7 @@ def export_models(
     output_names: Optional[List[str]] = None,
     device: str = "cpu",
     input_shapes: Optional[Dict] = None,
+    disable_dynamic_axes_fix: Optional[bool] = False,
```

Comment:
Why is this needed (could you precise in the docstring)?

Reply:
Correcting the dynamic axes gave a core dumped error when running onnxruntime.InferenceSession. And it is not necessary for Trainer, so I added the option to not correct it.

Reply:
It happens when export a ORT trained model with task `causal-lm-with-past` and only for the with-past decoder.
It tried automatically to fix the dynamic axes name of `logits dim=1`, and error occurs in this line:
https://github.com/huggingface/optimum/blob/4100c384127995de772a7a5b4a22df3b27e27e73/optimum/exporters/onnx/base.py#L335
I doubted it was the past-key-values as it only appeared for with past decoder and I checked,  but the dummy pkv look correct to me. Search a little bit, seems that this type of error could appear when reload model under different device / opset etc... I wonder if it is due to the fact that ORT trained model will have some Microsoft domain op while exporting to ONNX and `SessionOptions` need to precisely configured, but I did not dig further as it is special case and not impacting the ORTTrainer anyway.